eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala

```diff
@@ -95,15 +98,19 @@ class Switchboard(nodeParams: NodeParams, peerFactory: Switchboard.PeerFactory)
       // if this is an incoming connection, we might not yet have created the peer
       val peer = createOrGetPeer(authenticated.remoteNodeId, offlineChannels = Set.empty)
       val features = nodeParams.initFeaturesFor(authenticated.remoteNodeId)
+      val hasChannels = peersWithChannels.contains(authenticated.remoteNodeId)
       // if the peer is whitelisted, we sync with them, otherwise we only sync with peers with whom we have at least one channel
-      val doSync = nodeParams.syncWhitelist.contains(authenticated.remoteNodeId) || (nodeParams.syncWhitelist.isEmpty && peersWithChannels.contains(authenticated.remoteNodeId))
+      val doSync = nodeParams.syncWhitelist.contains(authenticated.remoteNodeId) || (nodeParams.syncWhitelist.isEmpty && hasChannels)
       authenticated.peerConnection ! PeerConnection.InitializeConnection(peer, nodeParams.chainHash, features, doSync)
+      if (!hasChannels && !authenticated.outgoing) {
+        incomingConnectionsTracker ! TrackIncomingConnection(authenticated.remoteNodeId)
+      }
 
     case ChannelIdAssigned(_, remoteNodeId, _, _) => context.become(normal(peersWithChannels + remoteNodeId))
 
     case LastChannelClosed(_, remoteNodeId) => context.become(normal(peersWithChannels - remoteNodeId))
 
-    case GetPeers => sender() ! context.children
+    case GetPeers => sender() ! context.children.filterNot(_.path.name.startsWith("incoming-connections-tracker"))
```

Comment:
```suggestion
    case GetPeers => sender() ! context.children.filterNot(_ == incomingConnectionsTracker.toClassic)
```

Reply:
I also added a test to `SwitchboardSpec` to detect if any other other child actor gets created by `Switchboard` and returned by `GetPeers`.